### PR TITLE
Add SIG Architecture Code Organization spotlight article

### PR DESCRIPTION
**Pull request by fsmunoz**
SIG Architecture: Code Organization spotlight article.

Closes: #482

### CODE REVIEW

File: content/en/blog/2024/sig-arch-code-spotlight.md
Comment (nimbinatus) [inline content/en/blog/2024/sig-arch-code-spotlight.md:118]:
```suggestion
take the opportunity to specifically highlight the work of [Tim Hockin](https://github.com/thockin)
```

File: content/en/blog/2024/sig-arch-code-spotlight.md
Comment (nimbinatus) [inline content/en/blog/2024/sig-arch-code-spotlight.md:129]: Any reason not to make the Slack link an actual link?
```suggestion
[slack.k8s.io](https://slack.k8s.io), and after that join the `#k8s-code-organization` channel. There is also a
```

**Review by nimbinatus — CHANGES_REQUESTED** [review]
Not going through a detailed grammar review after last time :) Left a couple questions and fixed a spelling error. All links have been checked.

File: content/en/blog/2024/sig-arch-code-spotlight.md
Comment (sftim) [inline content/en/blog/2024/sig-arch-code-spotlight.md:138]: (nit)
```suggestion
bot, so don’t hesitate to get involved in the `#k8s-code-organization` Slack channel.
```

File: content/en/blog/2024/sig-arch-code-spotlight.md
Comment (sftim) [inline content/en/blog/2024/sig-arch-code-spotlight.md:120]: Could instead link to [`/blog/2024/03/19/go-workspaces-in-kubernetes/`](https://k8s.dev/blog/2024/03/19/go-workspaces-in-kubernetes/), the article about this.

File: content/en/blog/2024/sig-arch-code-spotlight.md
Comment (sftim) [inline content/en/blog/2024/sig-arch-code-spotlight.md:34]: We an use the actual math symbol:
```suggestion
Kubernetes codebase. We have ≅2.2 million lines of Go code (which is steadily decreasing thanks to
```

File: content/en/blog/2024/sig-arch-code-spotlight.md
Comment (sftim) [inline content/en/blog/2024/sig-arch-code-spotlight.md:13]: (nit)
```suggestion
In this SIG Architecture spotlight I talked with [Madhav
```
There were two people talking, one was Madhav, so you'd use first person singular.

**Review by sftim — COMMENTED** [review]
LGTM apart from publication date.

Very small nits.

/hold
for assignment of publication date.

How about Thursday (2024-04-11)?

File: content/en/blog/2024/sig-arch-code-spotlight.md
Comment (fsmunoz) [reply content/en/blog/2024/sig-arch-code-spotlight.md:11]: Done.

File: content/en/blog/2024/sig-arch-code-spotlight.md
Comment (fsmunoz) [reply content/en/blog/2024/sig-arch-code-spotlight.md:13]: Yes, old habits that are not adequate here.

File: content/en/blog/2024/sig-arch-code-spotlight.md
Comment (fsmunoz) [reply content/en/blog/2024/sig-arch-code-spotlight.md:120]: Yes much better. Changed in last commit.

**Comment by fsmunoz** [comment]
@nimbinatus @sftim I think this is good to go, I've reflected all suggestions, updated the publication date for the suggested one, and squashed it in case it really is the final version.

**Comment by sftim** [comment]
@dims want to send in a PR to do a fixup here too?

(wrt https://github.com/kubernetes/website/pull/45853)

**Comment by dims** [comment]
may i defer to @fsmunoz please?

**Comment by fsmunoz** [comment]
@dims thanks for the fix, I can add it here if it helps, sure.

**Pull request merged**
